fs/osfs.py

```diff
@@ -584,9 +584,14 @@ def getsyspath(self, path):
 
     def geturl(self, path, purpose="download"):
         # type: (Text, Text) -> Text
-        if purpose != "download":
+        sys_path = self.getsyspath(path)
+        if purpose == "download":
+            return "file://" + self.getsyspath(path)
```

Comment:
Should the "download" purpose also be using `url_quote`, e.g. in case the syspath contains a space character?

Reply:
addressed your second concern!

to quote the url for "download", I would leave it to #343. I would like to keep the Pandora box closed in this PR. I opened it by accident and regretted it because it will take forever to get this PR merged.